demisto_sdk/commands/test_content/TestContentClasses.py

```diff
@@ -270,14 +273,44 @@ def configure_integrations(self, client: DefaultApi) -> bool:
             configured_integrations.append(integration)
         return True
 
-    def disable_integrations(self, client: DefaultApi) -> None:
+    def disable_integrations(self, client: DefaultApi, server_context: 'ServerContext') -> None:
         """
         Disables all integrations that the playbook uses
+        Clears server configurations set for the integration if there are such
+        Reset containers if server configurations were cleared
+
         Args:
             client: The demisto_client to use
+            server_context (ServerContext): The ServerContext instance in which the TestContext instance is created in
         """
         for integration in self.integrations:
             integration.disable_integration_instance(client)
+        updated_keys = self._clear_server_keys(client)
+        if updated_keys:
+            server_context._reset_containers()
+
+    def _clear_server_keys(self, client: DefaultApi) -> bool:
```

Comment:
Is there a chance that the server keys configured for a specific integration will be using key that should have non empty value for the rest of the build?
such as `python.pass.extra.keys` or any of the sort?

if there is no such case - then it's LGTM.

But if there is - maybe it's better to modify the `update_server_configuration` to return the previous configuration and cache them in the server_context somewhere and the use them when cleaning the configuration, otherwise you might clean necessary conf.

Reply:
as discussed, i think we're good here as empty string is the default taken on the backend